python/pyspark/sql/tests/test_udtf.py

```diff
@@ -1064,6 +1074,71 @@ def eval(self, row: Row):
         func = udtf(TestUDTF, returnType="a: int")
         return func
 
+    def test_df_asTable_chaining_methods(self):
+        class TestUDTF:
+            def eval(self, row: Row):
+                yield row["key"], row["value"]
+
+            def terminate(self):
+                if False:
+                    yield
+
+        func = udtf(TestUDTF, returnType="key: int, value: string")
+        df = self.spark.createDataFrame([(1, "a"), (1, "b"), (2, "c"), (2, "d")], ["key", "value"])
+        assertDataFrameEqual(
+            func(df.asTable().orderBy(df.value)),
```

Comment:
Wha't the behavior of this? In SQL, `order by` only is not allowed, IIRC. cc @dtenedor

Reply:
Similarly, what happens if, e.g., `func(df.asTable().partitionBy(df.key).orderBy(df.value)).partitionBy()`?

Reply:
It is allowed here
```
>>> TestUDTF(df.asTable().partitionBy("id").orderBy("id").partitionBy()).show()
+---+
|  a|
+---+
|  6|
|  7|
+---+
```

Reply:
Wha't the behavior of this?

Reply:
`order by` only is not supported in SQL as

```
[PARSE_SYNTAX_ERROR] Syntax error at or near 'ORDER'. SQLSTATE: 42601 (line 1, pos 59)

== SQL ==
SELECT * FROM test_udtf(TABLE (SELECT id FROM range(0, 8)) ORDER BY id)
-----------------------------------------------------------^^^
```

let me adjust that.

Reply:
`partition by` `order by` `partition by` is also not supported as

```
[PARSE_SYNTAX_ERROR] Syntax error at or near 'PARTITION'. SQLSTATE: 42601 (line 1, pos 87)

== SQL ==

SELECT * FROM test_udtf(TABLE (SELECT id FROM range(0, 8)) PARTITION BY id ORDER BY id PARTITION BY id)
---------------------------------------------------------------------------------------^^^
```

let me adjust that